mtags/src/main/scala/scala/meta/internal/pc/ScriptFirstImportPosition.scala

```diff
@@ -0,0 +1,103 @@
+package scala.meta.internal.pc
+
+import scala.annotation.tailrec
+
+import scala.meta._
+
+/**
+ * Used to determine the position for the first import for scala-cli `.scala` and `.sc` files.
+ * For scala-cli sources we need to skip `//> using` comments.
+ *
+ * For `.sc` Ammonite and Scala-Cli wraps the code for such files.
+ * The following code:
+ * ```scala
+ * val a = 1
+ * ```
+ * Is trasnformed and passed into PC as:
+ * ```scala
+ * ${tool-defauls-imports}
+ * object ${wrapperObject} {
+ * /*<${scriptMarker}>*/
+ *   val a = 1  <-- actual code
+ * }
+ * ```
+ * To find the proper position we need to find the object that contains `/*<${scriptMarker}>*/`
+ */
+object ScriptFirstImportPosition {
+
+  val usingDirectives: List[String] = List("// using", "//> using")
+
+  def ammoniteScStartOffset(text: String): Option[Int] = {
+    val it = tokenize(text).iterator
+    startMarkerOffset(it, "/*<start>*/").map(_ + 1)
+  }
+
+  def scalaCliScStartOffset(text: String): Option[Int] = {
+    val iterator = tokenize(text).iterator
+    startMarkerOffset(iterator, "/*<script>*/").map { startOffset =>
+      val offset =
+        skipUsingDirectivesOffset(iterator, None)
+          .getOrElse(startOffset)
+
+      offset + 1
+    }
+  }
```

Comment:
Can we run `skipUsingDirectivesOffset` anyway for both the ammonite script and scala-cli script?
I think it's ok to insert an import under using directive for ammonite script (using directive doesn't mean anything in Ammonite script though).

Reply:
We can. But there a might be a case with 0.00...1% chance that you have an ammonite source that just has a comment starting from using comment :smile_cat: 
Like:
```scala
// using this script ....
// bla-bla
${code}
```
+ it isn't costly to use `skipUsingDirectives` only scala-cli sc